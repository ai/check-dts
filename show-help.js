let { yellow: y, bold: b } = require('colorette')

module.exports = function showHelp (print) {
  print(
    b('Usage: ') + 'npx check-dts [file1 file2...]',
    'Check `.d.ts` files in open source library according to types tests',
    '',
    b('Arguments:'),
    '  ' +
      y('file#') +
      '       Optional a list of files/globs (otherwise matches all js/ts files)',
    '  ' + y('--version') + '   Show version',
    '  ' + y('--help') + '      Show this message'
  )
}
