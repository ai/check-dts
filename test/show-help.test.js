import { showHelp } from '../show-help.js'

function createStdout() {
  let result = { out: '' }
  result.print = (...lines) => {
    result.out += lines.join('\n') + '\n'
  }
  return result
}

it('prints version', () => {
  let stdout = createStdout()
  showHelp(stdout.print)
  expect(stdout.out).toMatchSnapshot()
})
