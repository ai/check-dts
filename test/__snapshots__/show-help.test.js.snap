// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prints version 1`] = `
"[1mUsage: [22mnpx check-dts [FILES]
Check \`.d.ts\` files in open source library according to types tests

[1mArguments:[22m
  [33m[FILES][39m    Optional a list of files/globs
  [33m--version[39m  Show version
  [33m--help[39m     Show this message
"
`;
